### PR TITLE
fix(material/slider): fix internal focus state on safari

### PR DESCRIPTION
**Pull request by wagnermaciel**
* Fixes https://github.com/angular/components/issues/28211

### CODE REVIEW

File: src/material/slider/slider-input.ts [hunk 1]
```diff
@@ -215,6 +215,11 @@ export class MatSliderThumb implements _MatSliderThumb, OnDestroy, ControlValueA
 
   /** Used to relay updates to _isFocused to the slider visual thumbs. */
   private _setIsFocused(v: boolean): void {
+    // The slider never receives focus in safari.
+    if (this._platform.SAFARI) {
```
Comment (crisbeto) [inline src/material/slider/slider-input.ts:219]: I'm not sure that we can do this. My understanding is that users have to enable focus in Safari.

Comment (wagnermaciel) [reply src/material/slider/slider-input.ts:219]: How do you enable this? I can't find a way to enable focus on `input type="range"` on safari

Comment (crisbeto) [reply src/material/slider/slider-input.ts:219]: So there's a setting at `Safari > Settings > Advanced > Press tab to highlight each item on the page` which I believe toggles whether things are focusable.

Comment (wagnermaciel) [reply src/material/slider/slider-input.ts]: I pushed a new approach which avoids breaking this edge case. One thing I noticed is that even with this setting enabled, the slider still does not maintain focus from pointer events

**Review by crisbeto — APPROVED** [review]

**Pull request merged**